bin/rust/rust_belt/lifetime_logic.gh

```diff
@@ -42,7 +45,17 @@ lemma void close_lifetime_intersection_dead_token(lifetime_t k, lifetime_t k1);
     requires [_]lifetime_dead_token(k);
     ensures [_]lifetime_dead_token(lifetime_intersection(k, k1));
 
-predicate thread_token(thread_id_t t); // Non-atomic token with Top mask ([NaInv: t.Top] in RustBelt)
+fixpoint void *Nshr() { return (void *) 0; }
+fixpoint void *Nrust() { return (void *) 1; }
+
+predicate thread_token(thread_id_t t, mask_t m);
+lemma void thread_token_split(thread_id_t t, mask_t m , mask_t to_split);
+    //@ requires thread_token(t, m) &*& mask_is_sub(to_split, m) == true;
+    //@ ensures thread_token(t, mask_diff(m, to_split)) &*& thread_token(t, to_split);
+
+lemma void thread_token_merge(thread_id_t t, mask_t m , mask_t m1);
+    //@ requires thread_token(t, m) &*& thread_token(t, m1) &*& mask_is_empty(mask_isect(m, m1)) == true;
```

Comment:
I'm pretty sure that this is_empty requirement is superfluous, so we should drop it. Does it not follow from `thread_token(t, m) &*& thread_token(t, m1)` that `mask_is_empty(mask_isect(m, m1))`?

Reply:
It should follow. I simply put it there because of the disjoint union in NAInv-tok-split rule in RustBelt.